lib/plsql/package.rb

```diff
@@ -1,29 +1,34 @@
 module PLSQL
 
   module PackageClassMethods #:nodoc:
-    def find(schema, package)
-      if schema.select_first(
-            "SELECT object_name FROM all_objects
-            WHERE owner = :owner
-              AND object_name = :package
-              AND object_type = 'PACKAGE'",
-            schema.schema_name, package.to_s.upcase)
-        new(schema, package)
-      # search for synonym
-      elsif (row = schema.select_first(
-            "SELECT o.owner, o.object_name
-            FROM all_synonyms s, all_objects o
-            WHERE s.owner IN (:owner, 'PUBLIC')
-              AND s.synonym_name = :synonym_name
-              AND o.owner = s.table_owner
-              AND o.object_name = s.table_name
-              AND o.object_type = 'PACKAGE'
-            ORDER BY DECODE(s.owner, 'PUBLIC', 1, 0)",
-            schema.schema_name, package.to_s.upcase))
-        new(schema, row[1], row[0])
-      else
-        nil
-      end
+    def find(schema, package_name)
+      find_package_in_schema(schema, package_name) || find_package_by_synonym(schema, package_name)
+    end
+
+    def find_package_in_schema(schema, package_name)
+      row = schema.select_first(<<-SQL, schema.schema_name, package_name.to_s.upcase)
```

Comment:
Can you think of a way to for package_name method parameter to be already passed in a valid form?
The "package_name" seems confusing, as it is a symbol and needs to be converted to string and upcase'd.